docs/DeveloperGuide.md

```diff
@@ -378,6 +413,96 @@ it will be useful for them to interact with their contact lists through the cust
+obtaining the tag's name, value and supervisor flag have been omitted from the diagram below.
+
+![](images/TagParserSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `ProfileCommandParser` and `ProfileCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+After the creation of the `TagCommand`, it will be executed by `LogicManager`. Below is the sequence diagram for the execution 
+of the `TagCommand`.
+
+![](images/TagSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `ProfileCommandParser` and `ProfileCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+**Step 1.** `LogicManager` executes the `TagCommand`.
+
+**Step 2.** `TagCommand` obtains the list of `Person`s corresponding to its range.
+
+**Step 3.** For each person in the obtained person list a new tag is generated by generateNewTag() and added to the person's list of tags (or used to replace the existing 
+tag if the person already has the tag). The new set of tags is used to instantiate the `editedPerson`. Details of adding the tag to the `editedPerson` has been 
+omitted for brevity.
+
+**Step 4.** The updated `editedPerson` is used to replace the original `person` with `setPerson`.
+
+#### Design Considerations
+
+Tags are used extensively in Teletubbies with a variety of purposes. Tags in Teletubbies are much more versatile than their
+implementation in AB-3. Tags now have both names and values, and can also be set to be only editable by supervisors. This
+allows for supervisors to create assignee tags for example, with name 'assignee' and value as the name of the assignee. 
+The tag can also be set to be modifiable only by supervisors, so that employees won't be able to tamper with the manpower 
+assignment. 
+
+The completion status of a contact is also internally represented as a tag, with name 'CompletionStatus' and value
+either 'INCOMPLETE', 'ONGOING' or 'COMPLETE'. Since the `filter` and `export` commands make use of tags, this allows users to filter and
+export contacts by 'CompletionStatus', since CompletionStatus is also a tag after all. Supervisors can also export contacts corresponding
+to a particular assignee for contact dissemination, which is a critical feature for supervisors.
+
+**Aspect: How general (in terms of functionality) should tags be?**
+
+* **Alternative 1 (current choice):** Users are free to set tag name, value and accessibility
+    * Pros: 
+      * Users have more configurability options for tags, which opens up use-cases and ultimately makes tags more useful.
+      * Implementation for `filter` and `export` is simplified since only tags will be used to specifying contacts for these commands.
```

Comment:
```suggestion
      * Implementation for `filter` and `export` is simplified since only tags will be used to specify contacts for these commands.
```